slsim/Sources/source.py

```diff
@@ -25,14 +26,23 @@ def __init__(self, source_dict, variability_model=None, kwargs_variab=None):
             kwargs_variab_extracted = {}
             for element in kwargs_variab:
                 if element in self.source_dict.colnames:
-                    kwargs_variab_extracted[element] = self.source_dict[element]
-                    self.variability_class = Variability(
-                        variability_model, **kwargs_variab_extracted
-                    )
+                    if (
+                        isinstance(self.source_dict[element], np.ndarray)
+                        and self.source_dict[element].ndim == 2
+                        and self.source_dict[element].shape[0] == 1
+                    ):
+                        kwargs_variab_extracted[element] = self.source_dict[
+                            element
+                        ].reshape(-1)
+                    else:
+                        kwargs_variab_extracted[element] = self.source_dict[element]
                 else:
                     raise ValueError(
                         "given keywords are not in the provided source catalog."
```

Comment:
can you also print the actual keyword in the raise statement? you can do that with e.g. "the variable %s is not found" % element